File: Persintencia_Python/src/AttendanceManagement.py
from typing import List
from include.Student import Student
from include.Course import Course
import os.path

class AttendanceManagement:
    def __init__(self):
        pass
    
    def takeAttendance(self, student: Student, course: Course, datetime: str, state: bool) -> None:
        fileExists = os.path.isfile("attendance.csv")
        with open("attendance.csv", mode="a") as csvFile:
            if not fileExists:
                csvFile.write("Id de Estudiantes, Nombre/s, Apellido/s, Id de Materia, Marca temporal, Presente\n")               
            csvFile.write(f"{student.getIdentifier()}, {student.getName()}, {student.getSurname()}, {course.getIdentifier()}, {datetime}, {state}\n")
            
    def showAttendance(self) -> None:
        if os.path.isfile("attendance.csv"):
            with open("attendance.csv") as csvFile:
                print(csvFile.read())

File: Persistencia_Cpp/src/AttendanceManagement.cpp
#include "../include/AttendanceManagement.hpp"

// Constructor
AttendanceManagement::AttendanceManagement()
{

}

void AttendanceManagement::takeAttendance(Student* student, Course* course, string datetime, bool state)
{

    ifstream testOpenFile("attendance.csv");
    bool fileExists = testOpenFile.good();
    testOpenFile.close();

    ofstream csvFile;
    csvFile.open("attendance.csv", ofstream::app);

    if(!fileExists)
    {

        csvFile << "Id de Estudiantes, "
                << "Nombre/s, "
                << "Apellido/s, "
                << "Id de Materia, "
                << "Marca temporal, "
                << "Presente\n";
    }

    csvFile << student->getIdentifier() << ", "
            << student->getName() << ", "
            << student->getSurname() << ", "
            << course->getIdentifier() << ", "
            << course->getName() << ", "
            << datetime << ", "
            << state << "\n";

    csvFile.close();
}

void AttendanceManagement::showAttendance()
{

    ifstream csvFile("attendance.csv");

    if(csvFile.is_open())
    {
        // Toma el contenido del csv para mostrarlo en la salida estandar
        cout << csvFile.rdbuf();
    }
        
    csvFile.close();
}

// Destructor
AttendanceManagement::~AttendanceManagement()
{

}